openrota/openrota-database-template.yaml

```diff
@@ -0,0 +1,213 @@
+
+         Username: ${POSTGRESQL_USER}
+         Password: ${POSTGRESQL_PASSWORD}
+    Database Name: ${POSTGRESQL_DATABASE}
+   Connection URL: postgresql://${DATABASE_SERVICE_NAME}:5432/
+
+  For more information about using this template, including OpenShift considerations, see https://github.com/sclorg/postgresql-container/.
+metadata:
+  name: openrota-database-template
+objects:
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${DATABASE_SERVICE_NAME}
+    labels:
+      tier: database
+  stringData:
+    database-name: ${POSTGRESQL_DATABASE}
+    database-password: ${POSTGRESQL_PASSWORD}
+    database-user: ${POSTGRESQL_USERNAME}
+
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: ${DATABASE_SERVICE_NAME}
+    labels:
+      tier: database
+  spec:
+    ports:
+      - name: postgresql
+        nodePort: 0
+        port: 5432
+        protocol: TCP
+        targetPort: 5432
+    selector:
+      name: ${DATABASE_SERVICE_NAME}
+    sessionAffinity: None
+    type: ClusterIP
+  status:
+    loadBalancer: {}
+
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    annotations:
+      kubernetes.io/reclaimPolicy: Delete
+    name: ${DATABASE_SERVICE_NAME}
+    labels:
+      tier: database
+  spec:
+    storageClassName: aws-ebs
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: ${VOLUME_CAPACITY}
+
+- apiVersion: apps/v1
+  kind: StatefulSet
+  metadata:
+    name: ${DATABASE_SERVICE_NAME}
+    labels:
+      tier: database
+  spec:
+    updateStrategy:
+      type: RollingUpdate
+    replicas: 2
+    selector:
+      matchLabels:
+        app: ${DATABASE_SERVICE_NAME}
+    serviceName: ${DATABASE_SERVICE_NAME}
+    template:
+      metadata:
+        labels:
+          app: ${DATABASE_SERVICE_NAME}
+      spec:
+        containers:
+          - capabilities: { }
+            env:
+              - name: POSTGRESQL_USER
+                valueFrom:
+                  secretKeyRef:
+                    key: database-user
+                    name: ${DATABASE_SERVICE_NAME}
+              - name: POSTGRESQL_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    key: database-password
+                    name: ${DATABASE_SERVICE_NAME}
+              - name: POSTGRESQL_DATABASE
+                valueFrom:
+                  secretKeyRef:
+                    key: database-name
+                    name: ${DATABASE_SERVICE_NAME}
+              - name: PGDATA
+                valueFrom:
+                  secretKeyRef:
+                    key: pgdata
+                    name: ${DATABASE_SERVICE_NAME}
+              - name: POD_IP
+                valueFrom:
+                  fieldRef:
+                    apiVersion: v1
+                    fieldPath: status.podIP
+            image: ' '
+            imagePullPolicy: IfNotPresent
+            livenessProbe:
+              exec:
+                command:
```

Comment:
Changed the livenessProbe and readinessProbe command w.r.t testing on my OCP 4.10 cluster. We can revert this if you face problems.

Reply:
not sure what ocp version we are on MP+. We should use the preprod cluster and test the compatibility.

Reply:
I will deploy to preprod cluster tomorrow and let you know if we are fine with that.